CppSamples/Layers/ConfigureElectronicNavigationalCharts/ConfigureElectronicNavigationalCharts.cpp

```diff
@@ -0,0 +1,338 @@
+  {
+    return;
+  }
+  m_colorScheme = colorScheme;
+  updateDisplaySettings();
+  emit colorSchemeChanged();
+}
+
+EncAreaSymbolizationType ConfigureElectronicNavigationalCharts::areaSymbolizationType() const
+{
+  return m_areaSymbolizationType;
+}
+
+void ConfigureElectronicNavigationalCharts::setAreaSymbolizationType(Esri::ArcGISRuntime::EncAreaSymbolizationType areaSymbolizationType)
+{
+  if (m_areaSymbolizationType == areaSymbolizationType)
+  {
+    return;
+  }
+  m_areaSymbolizationType = areaSymbolizationType;
+  updateDisplaySettings();
+  emit areaSymbolizationTypeChanged();
+}
+
+EncPointSymbolizationType ConfigureElectronicNavigationalCharts::pointSymbolizationType() const
+{
+  return m_pointSymbolizationType;
+}
+
+void ConfigureElectronicNavigationalCharts::setPointSymbolizationType(Esri::ArcGISRuntime::EncPointSymbolizationType pointSymbolizationType)
+{
+  if (m_pointSymbolizationType == pointSymbolizationType)
+  {
+    return;
+  }
+  m_pointSymbolizationType = pointSymbolizationType;
+  updateDisplaySettings();
+  emit pointSymbolizationTypeChanged();
+}
+
+void ConfigureElectronicNavigationalCharts::updateDisplaySettings()
+{
+  // Hold a reference to the application-wide ENC Display Settings
+  EncDisplaySettings* globalDisplaySettings = EncEnvironmentSettings::displaySettings();
+
+  // Hold a reference to the application-wide ENC Mariner Settings (part of display settings)
+  EncMarinerSettings* globalMarinerSettings = globalDisplaySettings->marinerSettings();
+
+  // Directly apply stored enum values
+  globalMarinerSettings->setColorScheme(m_colorScheme);
+  globalMarinerSettings->setAreaSymbolizationType(m_areaSymbolizationType);
+  globalMarinerSettings->setPointSymbolizationType(m_pointSymbolizationType);
+}
+
+void ConfigureElectronicNavigationalCharts::clearAllSelections()
+{
+  if (!m_mapView)
+  {
+    return;
+  }
+
+  // For each layer in the operational layers that is an ENC layer
+  LayerListModel* operationalLayers = m_map->operationalLayers();
+  for (QObject* obj : *operationalLayers)
+  {
+    if (EncLayer* encLayer = qobject_cast<EncLayer*>(obj); encLayer)
+    {
+      encLayer->clearSelection();
+    }
+  }
+
+  // Hide the callout
+  m_mapView->calloutData()->setVisible(false);
+}
+
+void ConfigureElectronicNavigationalCharts::onGeoViewTapped(QMouseEvent& mouseEvent)
+{
+  if (!m_mapView)
+  {
+    return;
+  }
+
+  // Set callout position for new location
+  const Point mapPoint = m_mapView->screenToLocation(mouseEvent.position().x(), mouseEvent.position().y());
+  m_mapView->calloutData()->setLocation(mapPoint);
+
+  // Clear existing selections
+  clearAllSelections();
+
+  // Perform the identify operation on ENC layers specifically
+  const double tolerance = 12.0;
+  const bool returnPopupsOnly = false;
+
+  m_mapView->identifyLayersAsync(mouseEvent.position(), tolerance, returnPopupsOnly)
+    .then(this,
+          [this](const QList<IdentifyLayerResult*>& results)
```

Comment:
I would consider putting this logic into a private helper. It's so long that by the time you scroll to the end, it's hard to tell what the `.onFailed` goes with.